code/datums/diseases/_disease.dm

```diff
@@ -86,11 +93,100 @@
 	else if(SPT_PROB(stage_prob*slowdown, seconds_per_tick))
 		update_stage(min(stage + 1, max_stages))
 
-	return !carrier
+	if(stage == max_stages && stage_peaked != TRUE) //mostly a sanity check in case we manually set a virus to max stages
+		stage_peaked = TRUE
+
+	if(stage_peaked && !(disease_flags & CHRONIC) && disease_flags & CURABLE && bypasses_immunity != TRUE)
+		if(stage == max_stages) //every cycle we spend at max stage counts towards eventually curing the virus
+			peaked_cycles += 1
+		switch(severity)
+			if(DISEASE_SEVERITY_POSITIVE) //good viruses don't go anywhere after hitting max stage - you can try to get rid of them by sleeping earlier
+				cycles_to_beat = DISEASE_CYCLES_POSITIVE
+				if(affected_mob.satiety > 0 || slowdown == 1) //any sort of malnourishment/immunosuppressant opens you to losing a good virus
```

Comment:
An unintended consequence of satiety is that TRAIT_NOHUNGER makes you not have any.
Ethereals actually require that trait for their hunger system to work, so they wouldn't be able to sustain any positive virus!
I'm not sure that's intended...